freqtrade/freqai/RL/BaseReinforcementLearningModel.py

```diff
@@ -236,18 +248,19 @@ def predict(
             unfiltered_df, dk.training_features_list, training_filter=False
         )
 
-        filtered_dataframe = self.drop_ohlc_from_df(filtered_dataframe, dk)
-
-        filtered_dataframe = dk.normalize_data_from_metadata(filtered_dataframe)
-        dk.data_dictionary["prediction_features"] = filtered_dataframe
+        dk.data_dictionary["prediction_features"] = self.drop_ohlc_from_df(filtered_dataframe, dk)
 
-        # optional additional data cleaning/analysis
-        self.data_cleaning_predict(dk)
+        dk.data_dictionary["prediction_features"], outliers, _ = dk.feature_pipeline.transform(
+            dk.data_dictionary["prediction_features"], outlier_check=True)
 
         pred_df = self.rl_model_predict(
             dk.data_dictionary["prediction_features"], dk, self.model)
         pred_df.fillna(0, inplace=True)
 
+        if self.freqai_info.get("DI_threshold", 0) > 0:
+            dk.DI_values = dk.feature_pipeline["di"].di_values
```

Comment:
Shouldn't this test be `if 'di' in dk.feature_pipeline:`? 

It'll work for regular cases - but assuming a user uses a config with `DI_threshold` set - but implements `define_data_pipeline` - then this can cause an implicit crash, as the configuration does no longer correspond to the generated pipeline.

Reply:
good catch, fixed.